examples/pet-mad/pet-mad.py

```diff
@@ -0,0 +1,541 @@
+"""
+The PET-MAD universal potential
+===============================
+
+:Authors: Philip Loche `@PicoCentauri <https://github.com/picocentauri>`_,
+          Michele Ceriotti `@ceriottm <https://github.com/ceriottm>`_,
+          Arslan Mazitov `@abmazitov <https://github.com/abmazitov>`_
+
+This example demonstrates how to use the PET-MAD model with ASE, `i-PI
+<https://ipi-code.org>`_ and `LAMMPS <https://lammps.org>`_. PET-MAD is a "universal"
+machine-learning forcefield trained on a dataset that aims to incorporate a very high
+degree of structural diversity.
+
+The point-edge transformer (PET) is an unconstrained architecture that achieves a high
+degree of symmetry compliance through data augmentation during training (see the `PET
+paper
+<https://proceedings.neurips.cc/paper_files/paper/2023/file/fb4a7e3522363907b26a86cc5be627ac-Paper-Conference.pdf>`_
+for more details). The unconstrained nature of the model simplifies its implementation
+and structure, making it computationally efficient and very expressive.
+
+The MAD dataset combines "stable" inorganic structures from the MC3D dataset, 2D
+structures from the MC2D dataset, and molecular crystals from the ShiftML dataset with
+"Maximum Atomic Diversity" configurations, generated by distorting the composition and
+structure of these stable templates. By doing so, PET-MAD achieves state-of-the-art
+accuracy despite the MAD dataset containing fewer than 100k structures. The reference
+DFT settings are highly converged, but limited to a PBEsol functional, so the accuracy
+against experimental data depends on how good this level of theory is for a given
+system. PET-MAD is introduced, and benchmarked for several challenging modeling tasks,
+in `this preprint <http://arxiv.org/TBD>`_.
+"""
+
+# %%
+#
+# Start by importing the required libraries.
+
+import os
+import subprocess
+from copy import copy, deepcopy
+
+# ASE and i-PI scripting utilities
+import ase.units
+import chemiscope
+import matplotlib.pyplot as plt
+import numpy as np
+import requests
+from ase.optimize import LBFGS
+from ipi.utils.mathtools import get_rotation_quadrature_lebedev
+from ipi.utils.parsing import read_output, read_trajectory
+from ipi.utils.scripting import (
+    InteractiveSimulation,
+    forcefield_xml,
+    motion_nvt_xml,
+    simulation_xml,
+)
+
+# metatensor models
+from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+from metatrain.utils.io import load_model
+
+
+if hasattr(__import__("builtins"), "get_ipython"):
+    get_ipython().run_line_magic("matplotlib", "inline")  # noqa: F821
+
+
+# %%
+# Inference on the MAD test set
+# -----------------------------
+#
+# We begin by using the ``ase``-compatible calculator to evaluate energy and forces for
+# a test dataset that contains both hold-out structures from the MAD dataset, and a few
+# structures from popular datasets (MPtrj, Alexandria, OC2020, SPICE, MD22) re-computed
```

Comment:
should this contains links to the datasets?

Reply:
added!